x-pack/plugins/security/server/lib/authentication/providers/oidc.ts

```diff
@@ -113,11 +111,22 @@ function isAccessTokenExpiredError(err?: any) {
  * Provider that supports authentication using an OpenID Connect realm in Elasticsearch.
  */
 export class OIDCAuthenticationProvider extends BaseAuthenticationProvider {
-  constructor(protected readonly options: OIDCProviderOptions) {
+  private readonly realm: string;
+
+  constructor(
+    protected readonly options: Readonly<AuthenticationProviderOptions>,
+    oidcOptions: Readonly<AuthenticationProviderSpecificOptions>
+  ) {
     super(options);
-    if (!this.options.realm) {
+    if (!oidcOptions.realm) {
       throw new Error('Realm name must be specified');
     }
+
+    if (type(oidcOptions.realm) !== 'string') {
```

Comment:
Not a criticism, just a question: what does this give us over the built in typeof operator?

Reply:
For string type checking, we could just use typeof directly, it mainly comes in handy for those “gotcha types”.